File: apis/src/main/java/com/beat/apis/config/WebConverterConfig.java
package com.beat.apis.config;

import com.beat.apis.config.converter.StringToEnumCustomConverterFactory;
import org.springframework.context.annotation.Configuration;
import org.springframework.format.FormatterRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration(proxyBeanMethods = false)
public class WebConverterConfig implements WebMvcConfigurer {

	@Override
	public void addFormatters(FormatterRegistry registry) {
		registry.addConverterFactory(new StringToEnumCustomConverterFactory());
	}
}

File: apis/src/main/kotlin/com/beat/apis/config/WebConverterConfig.kt
package com.beat.apis.config

import com.beat.apis.web.converter.CaseInsensitiveStringToEnumConverterFactory
import org.springframework.context.annotation.Configuration
import org.springframework.format.FormatterRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration(proxyBeanMethods = false)
class WebConverterConfig : WebMvcConfigurer {
    override fun addFormatters(registry: FormatterRegistry) {
        registry.addConverterFactory(CaseInsensitiveStringToEnumConverterFactory())
    }
}
